core/lib/types/src/storage_writes_deduplicator.rs

```diff
@@ -107,46 +110,57 @@ impl StorageWritesDeduplicator {
                 &mut self.metrics.repeated_storage_writes
             };
 
+            let total_size = &mut self.metrics.total_updated_values_size;
+
             match (was_key_modified, modified_value) {
                 (true, None) => {
                     let value = self.modified_key_values.remove(&key).unwrap_or_else(|| {
                         panic!("tried removing key: {:?} before insertion", key)
                     });
                     *field_to_change -= 1;
+                    *total_size -= value.size;
                     updates.push(UpdateItem {
                         key,
                         update_type: UpdateType::Remove(value),
                         is_write_initial,
                     });
                 }
                 (true, Some(new_value)) => {
+                    let value_size = compress_with_best_strategy(initial_value, new_value).len();
                     let old_value = self
```

Comment:
Why there was a need to rewrite this part? I mean, 

```
                    self.modified_key_values.insert(
                        key,
                        ModifiedSlot {
                            value: new_value,
                            tx_index: log.log_query.tx_number_in_block,
                            size: value_size,
                        },
                    );
```

would already return you the old value

Reply:
i think it was left over from an intermediate change